src/babel-custom.js

```diff
@@ -0,0 +1,160 @@
+import { createConfigItem } from '@babel/core';
+import babelPlugin from 'rollup-plugin-babel';
+import merge from 'lodash.merge';
+import { isTruthy } from './utils';
+
+const ESMODULES_TARGET = {
+	esmodules: true,
+};
+
+const mergeConfigItems = (type, ...configItemsToMerge) => {
+	const mergedItems = [];
+
+	configItemsToMerge.forEach(configItemToMerge => {
+		configItemToMerge.forEach(item => {
+			const itemToMergeWithIndex = mergedItems.findIndex(
+				mergedItem => mergedItem.file.resolved === item.file.resolved,
+			);
+
+			if (itemToMergeWithIndex === -1) {
+				mergedItems.push(item);
+				return;
+			}
+
+			mergedItems[itemToMergeWithIndex] = createConfigItem(
+				[
+					mergedItems[itemToMergeWithIndex].file.resolved,
+					merge(mergedItems[itemToMergeWithIndex].options, item.options),
+				],
+				{
+					type,
+				},
+			);
+		});
+	});
+
+	return mergedItems;
+};
+
+const createConfigItems = (type, items) => {
+	return items.map(({ name, ...options }) => {
+		return createConfigItem([require.resolve(name), options], { type });
+	});
+};
+
+export const config = (config, { customOptions }) => {
+	const defaultPlugins = createConfigItems(
+		'plugin',
+		[
+			{
+				name: '@babel/plugin-transform-react-jsx',
+				pragma: customOptions.pragma || 'h',
+				pragmaFrag: customOptions.pragmaFrag || 'Fragment',
+			},
+			!customOptions.typescript && {
+				name: '@babel/plugin-transform-flow-strip-types',
+			},
+			isTruthy(customOptions.defines) && {
+				name: 'babel-plugin-transform-replace-expressions',
+				replace: customOptions.defines,
+			},
+			!customOptions.modern && {
+				name: 'babel-plugin-transform-async-to-promises',
+				inlineHelpers: true,
+				externalHelpers: true,
+			},
+			{
+				name: '@babel/plugin-proposal-class-properties',
+				loose: true,
+			},
+			!customOptions.modern && {
+				name: '@babel/plugin-transform-regenerator',
+				async: false,
+			},
+			{
+				name: 'babel-plugin-macros',
+			},
+		].filter(Boolean),
+	);
+
+	const babelOptions = config.options || {};
+
+	const envIdx = (babelOptions.presets || []).findIndex(preset =>
+		preset.file.request.includes('@babel/preset-env'),
+	);
+
+	if (envIdx !== -1) {
+		const preset = babelOptions.presets[envIdx];
+		babelOptions.presets[envIdx] = createConfigItem(
+			[
+				preset.file.resolved,
+				Object.assign(
+					merge(
+						{
+							loose: true,
+							useBuiltIns: false,
+							targets: customOptions.targets,
+						},
+						preset.options,
+						{
+							modules: customOptions.jest ? 'auto' : false,
```

Comment:
I added a `jest` flag here. I was not sure if I should reuse the `modern` flag here.